app/models/concerns/eligibility_checkable.rb

```diff
@@ -0,0 +1,122 @@
+module EligibilityCheckable
+  extend ActiveSupport::Concern
+
+  def status
+    if eligible_now?
+      :eligible_now
+    elsif eligible_later?
+      :eligible_later
+    elsif ineligible?
+      :ineligible
+    else
+      :undetermined
+    end
+  end
+
+  def ineligible?
+    common_ineligible_attributes? || specific_ineligible_attributes?
+  end
+
+  # It's not good to call either `eligible_now?` or `eligible_later?` directly
+  # because of the following business rule which is captured in the `status`
+  # method above: if a claim is both eligible now and eligible later, it's
+  # deemed eligible now. The eligible later option is *only* shown as a consolation
+  # prize if there's nothing eligible now.
+  #
+  # So if a claim is both eligible now and eligible later somebody
+  # might check one, find it to be true, and assume the other is false.
+  # This is why you see a lot of code checking if something is eligible later
+  # but not eligible now and vice versa. Avoid these conundrums by using the
+  # `status` method in this class instead which keeps the states mutually exclusive.
+  def eligible_now?
+    common_eligible_now_attributes? && specific_eligible_now_attributes?
+  end
+
+  # Avoid calling this method directly and use the `status` method instead.
+  def eligible_later?
+    common_eligible_later_attributes? && specific_eligible_later_attributes?
+  end
+
+  private
+
+  def common_ineligible_attributes?
+    [has_indicated_ineligible_school?, trainee_teacher?, supply_teacher_lacking_either_long_contract_or_direct_employment?, poor_performance?, no_selectable_subjects?, ineligible_cohort?, insufficient_teaching?].any?
+  end
+
+  def has_indicated_ineligible_school?
```

Comment:
The convention is that you don't usually add `has_` if the method ends in `?`. Same for `has_indicated_eligible_school?`.

Reply:
I'll do a bit of renaming in that class